packages/core/src/lib/markdown-it/highlight/Highlighter.ts

```diff
@@ -0,0 +1,23 @@
+import { Boundary, collateAllIntervals, splitCodeAndIndentation } from './helper';
+
+export class Highlighter {
+  static highlightWholeLine(codeStr: string) {
+    return `<span class="highlighted">${codeStr}\n</span>`;
```

Comment:
I think it's not necessary to name the variable `codeStr` since we know it's a string (due to the `: string` type annotation) - possible alternatives could be `code`, `line`, `lineToHighlight` or anything else you see fit :)

same for the `codeStr` and `dataStr` below!